include/highfive/H5DataType.hpp

```diff
@@ -80,7 +92,58 @@ class AtomicType : public DataType {
 
     typedef T basic_type;
 };
-}
+
+
+///
+/// \brief A structure representing a set of fixed-length strings
+///
+/// Although fixed-len arrays can be created 'raw' without the need for
+/// this structure, to retrieve results efficiently it must be used.
+///
+template <std::size_t N>
+class FixedLenStringArray {
```

Comment:
IMO the `begin`, `end`, `cbegin`, `cend`, `rbegin`, and `rend`, `at`, `front`, `back`, `empty`, and `size`, member methods would be useful for the user.

Reply:
I thought about inheriting instead of composing... still not sure the best option here

Reply:
I usually prefer encapsulation over inheritance when it comes to standard containers.
I prefer to hide complexity, and I think their destructors are not virtual.